src/OpenTK.Windowing.Desktop/NativeWindow.cs

```diff
@@ -519,6 +521,25 @@ public unsafe Vector2i MaximumSize
             }
         }
 
+        private (int numerator, int denominator) _aspectRatio;
+
+        /// <summary>
+        /// Gets or sets the aspect ratio of this window specified as a numerator and denominator.
+        /// </summary>
+        /// <remarks>
+        /// Set both the values to <c>-1</c> to disable aspect ratio locking.
+        /// If you set size limits and an aspect ratio that conflict, the results are undefined.
+        /// </remarks>
+        public unsafe (int numerator, int denominator) AspectRatio
+        {
+            get => _aspectRatio;
+            set
+            {
+                _aspectRatio = value;
+                GLFW.SetWindowAspectRatio(WindowPtr, value.numerator, value.denominator);
+            }
+        }
+
```

Comment:
I think we want to say that this locks the window to this aspect ratio, but otherwise I think this is fine.